packages/py-moose-lib/moose_lib/dmv2/olap_dictionary.py

```
@@ -0,0 +1,547 @@
+    """SQL query string for the local ClickHouse server."""
+    source_tables: Optional[list[Any]] = None
+    """Explicit dependency list — required when ``source_query`` is set."""
+    external_source: Optional[ExternalSource] = None
+    """External system source (MySQL, MongoDB, HTTP, etc.)."""
+
+    # ── Key and schema ────────────────────────────────────────────────────────
+    primary_key: list[str]
+    """List of primary-key column names (single for simple layouts, multi for COMPLEX_KEY_*)."""
+    layout: DictionaryLayout
+    """Memory layout / storage model (discriminated by ``type``)."""
+    lifetime: Union[int, DictionaryLifetime] = DictionaryLifetime(min=0, max=0)
```

Ruff (0.15.7)

[warning] Use `X | Y` for type annotations

Convert to `X | Y`

(UP007)